src/api/forms/service.js

```diff
@@ -133,6 +134,54 @@ export async function updateDraftFormDefinition(formId, formDefinition) {
   return draftFormDefinition.create(formId, formDefinition)
 }
 
+/**
+ * Creates the live form from the current draft state
+ * @param {string} formId - ID of the form
+ * @param {FormMetadataAuthor} author - the author of the new live state
+ */
+export async function createLiveFromDraft(formId, author) {
+  // Get the form metadata from the db
+  const form = await getForm(formId)
+
+  if (!form) {
+    throw Boom.notFound(`Form with id '${formId}' not found`)
+  }
+
+  if (!form.draft) {
+    throw Boom.notFound(`Form with id '${formId}' has no draft state`)
+  }
+
+  // Build the live state
+  const now = new Date()
+  const set = !form.live
+    ? {
+        // Initialise the live state
+        live: {
+          updatedAt: now,
+          updatedBy: author,
+          createdAt: now,
+          createdBy: author
+        }
+      }
+    : {
+        // Partially update the live state
+        'live.updatedAt': now,
+        'live.updatedBy': author
+      }
+
+  // Copy the draft form definition
+  await draftFormDefinition.createLiveFromDraft(formId)
+
+  // Update the form with the live state and clear the draft
+  const result = await formMetadata.update(formId, {
+    $set: set,
+    $unset: { draft: '' }
+  })
```

Comment:
Not a blocker, but every time LocalStack starts I'm reminded how separate these two are. Wondering where we should capture the error scenarios?

E.g. The live definition was created but the metadata update failed, vice versa

Reply:
I need to speak to @alexluckett about this. I'm not sure the value of storing definitions in S3 anymore. We could be using mongo and getting transactional writes across metadata and definitions.